chanfitness/chanevent.go

```diff
@@ -60,14 +60,36 @@ type chanEventLog struct {
 	closedAt time.Time
 }
 
+// newEventLog creates an event log for a channel. If the peer is online at
+// the time the event log is created, add an online event with the same
+// timestamp as the event log so that the event log does not have a gap
+// between open time and the online event in the case where the peer was
+// already online.
 func newEventLog(fundingTxID wire.OutPoint, peer route.Vertex,
-	now func() time.Time) *chanEventLog {
+	now func() time.Time, peerOnline bool) *chanEventLog {
 
-	return &chanEventLog{
+	// Set the opened time of the event log to the present.
+	openedAt := now()
+
+	eventlog := &chanEventLog{
 		fundingTxID: fundingTxID,
 		peer:        peer,
 		now:         now,
+		openedAt:    openedAt,
+	}
+
+	// If the peer is online at startup, add a peer online event with the same
+	// timestamp as the event log opening time.
+	if peerOnline {
```

Comment:
I would think that just setting `openedAt` here is enough to fix the bug? Or is the logging of the first event right here also required?

Reply:
Adding an online event on startup for online peers is not new behaviour, it just used to be separated into creating the event log then adding the event. This is required to have a starting state for online calculations (ie, knowing whether we started with the peer online or offline). If we do not log this event at the same time as the channel opening, all channels that exit at startup will present as if they had a brief offline period.

I will update the comment to be more clear.

Reply:
>all channels that exit at startup will present as if they had a brief offline period

Is this worth adding the alternative code path for? Is 100% online even possible at all? When `lnd` starts, not all channels are up all of a sudden. Maybe it is even more realistic to keep it disconnected.

Reply:
They aren't all up, but if the peer is online, they are being a good channel peer (which is what we are measuring), regardless of what our current state is. If/When we start to calculate peer flaps based on the number of offline periods the peer has, this will falsely report an extra flap for the peer as well. I think it's better to have some way of identifying that this is a different scenario. 

I could record an event regardless of state, if that makes it more consistent? Or add a bool on the eventLog, but I think that adds more code than this approach.

Reply:
Ok, so we are measuring uptime excluding our own down time? When this is moved to persistent storage, there will not be a off and on event when we shut down `lnd` ourselves?

I am fine with what you have here, it is not very important.

Reply:
Yes, I think the goal here is to measure peer uptime, since our downtime is the same for all channels.

Reply:
When we start the peer notifier, we'll get the peer online event again I guess. Does this mean we will have two consecutive `online` events for each peer in the beginning of the log?

Reply:
It does. I thought we were already listening for online events at this stage so it would be missed, but that is not the case. I'm going to remove this online start event, since we will record it once we start consuming peer online events. There will be a small gap but it looks a lot cleaner doing it this way, so I think it's worth living with it.

Reply:
I think that is fine for now. If it turns out we cannot live with the gap, we can address it later 😆